tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -98,8 +96,19 @@ public TracerSettings(IConfigurationSource source)
                          new ConcurrentDictionary<string, string>();
 
             // Filter out tags with empty keys or empty values, and trim whitespace
-            GlobalTags = GlobalTags.Where(kvp => !string.IsNullOrWhiteSpace(kvp.Key) && !string.IsNullOrWhiteSpace(kvp.Value))
-                                   .ToDictionary(kvp => kvp.Key.Trim(), kvp => kvp.Value.Trim());
+            if (GlobalTags.Count > 0)
+            {
+                var globalTags = new Dictionary<string, string>();
+                foreach (var kvp in GlobalTags)
+                {
+                    if (!string.IsNullOrWhiteSpace(kvp.Key) && !string.IsNullOrWhiteSpace(kvp.Value))
+                    {
+                        globalTags[kvp.Key.Trim()] = kvp.Value.Trim();
+                    }
+                }
+
+                GlobalTags = globalTags;
+            }
```

Comment:
And this is _undeniably_ uglier and harder to follow - the perf gains for this one-off method better be worth it 😛

Reply:
+1

Reply:
😢  harder? we rewrite IL in C++ in the same project and emit methods for ducktype 🤣🤣🤣🤣🤣